src/core/socket.c

```diff
@@ -1919,6 +1924,20 @@ static int socket_spawn(Socket *s, ExecCommand *c, PidRef *ret_pid) {
         if (r < 0)
                 return r;
 
+        if (s->pass_fds_to_exec) {
```

Comment:
Please add a comment here explaining that we don't pass fds to ExecStartPre= intentionally